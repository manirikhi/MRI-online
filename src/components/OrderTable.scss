.order-table {
  height: 400px;
  width: 100%
}

.order-label {
  padding: 1rem;
}