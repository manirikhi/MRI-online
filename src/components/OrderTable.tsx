import React, { useEffect } from 'react';
import {AgGridColumn, AgGridReact} from 'ag-grid-react';
import { fetchOrders } from '../store/reducers/orderReducer'
import './OrderTable.scss'
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
import makeSelectOrders, { orderLoadingSelector } from '../store/selectors/orderSelector';
import { useAppDispatch, useAppSelector } from '../store/hooks';

function OrderTable() {
    const dispatch = useAppDispatch()
    const orders= useAppSelector(makeSelectOrders);
    const loading= useAppSelector(orderLoadingSelector);
    

    useEffect(()=> {
      dispatch(fetchOrders())
    }, [dispatch])
    
    return (
        <div className="ag-theme-alpine order-table">
            <h3 className="order-label">Orders</h3>
            {loading ? 
                <p>Loading orders.....</p>:
            <AgGridReact
                rowSelection="multiple"
                pagination={true}
                rowData={orders}>
                <AgGridColumn field="orderNumber" sortable={ true } filter={ true } checkboxSelection={true} ></AgGridColumn>
                <AgGridColumn field="customerName" sortable={ true } filter={ true } ></AgGridColumn>
                <AgGridColumn field="customerAddress" sortable={ true } filter={ true } ></AgGridColumn>
                <AgGridColumn field="orderValue" sortable={ true } filter={ true } ></AgGridColumn>
                <AgGridColumn field="orderDate" sortable={ true } filter={ true } ></AgGridColumn>
                <AgGridColumn field="shipDate" sortable={ true } filter={ true } ></AgGridColumn>
                <AgGridColumn field="status" sortable={ true } filter={ true } ></AgGridColumn>
            </AgGridReact>
            }
        </div>
    );
}

export default OrderTable;
