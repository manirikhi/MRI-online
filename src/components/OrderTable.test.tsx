import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import OrderTable from "./OrderTable";
import { store } from "../store/store";

test("renders OrderTable with loading state", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <OrderTable />
    </Provider>
  );
  const orderLabel = screen.getByText("Orders");
  expect(orderLabel).toBeInTheDocument();
  const loading = screen.getByText("Loading orders.....");
  expect(loading).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test("renders OrderTable with table data", async () => {
  render(
    <Provider store={store}>
      <OrderTable />
    </Provider>
  );
  await waitFor(() => {
    const orderNumber = screen.getByText("Order Number");
    expect(orderNumber).toBeInTheDocument();
    const order = screen.getByText("100000");
    expect(order).toBeInTheDocument();
  });
});

test("renders OrderTable with table data", async () => {
  render(
    <Provider store={store}>
      <OrderTable />
    </Provider>
  );
  await waitFor(() => {
    const orderNumber = screen.getByText("Order Number");
    expect(orderNumber).toBeInTheDocument();
    const order = screen.getByText("100000");
    expect(order).toBeInTheDocument();
    fireEvent.click(orderNumber)
    expect(screen.getByText("100000")).toBeInTheDocument();
    fireEvent.click(orderNumber)
    expect(screen.getByText("100000")).toBeInTheDocument();
  });
});
