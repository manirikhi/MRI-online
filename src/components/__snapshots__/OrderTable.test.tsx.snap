// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders OrderTable with loading state 1`] = `
<DocumentFragment>
  <div
    class="ag-theme-alpine order-table"
  >
    <h3
      class="order-label"
    >
      Orders
    </h3>
    <p>
      Loading orders.....
    </p>
  </div>
</DocumentFragment>
`;
